### PR TITLE
Better diagnostics for dlltool errors.

### PR DESCRIPTION
**Pull request by jfgoog**
When dlltool fails, show the full command that was executed. In particular, llvm-dlltool is not very helpful, printing a generic usage message rather than what actually went wrong, so stdout and stderr aren't of much use when troubleshooting.

### CODE REVIEW

**Comment by jfgoog** [comment]
Example of output:

```
error: Dlltool could not create import library with /usr/local/google/home/jamesfarrell/src/rust-toolchain/prebuilts/clang/host/linux-x86/clang-r487747c/bin/llvm-dlltool -d /tmp/rustcNiX2JN/kernel32.def -D kernel32.dll -l /tmp/rustcNiX2JN/kernel32.lib -m i386:x86-64 -f --64 --no-leading-underscore --temp-prefix /tmp/rustcNiX2JN/kernel32.dll:
       OVERVIEW: llvm-dlltool

       USAGE: llvm-dlltool [options] file...

       OPTIONS:
         -D <value> Specify the input DLL Name
         -d <value> Input .def File
         -f <value> Assembler Flags
         -k         Kill @n Symbol from export
         -l <value> Generate an import lib
         -m <value> Set target machine
         -S <value> Assembler

       TARGETS: i386, i386:x86-64, arm, arm64
```

(Note that the underlying problem causing this particular error is that llvm-dlltool doesn't support --temp-prefix, which I intend to address separately.)

**Comment by fee1-dead** [comment]
I'm not familiar with this.

r? compiler

**Review by WaffleLapkin — APPROVED** [review]
Can you squash the commits together?

**Comment by jfgoog** [comment]
Squashed. Thanks for the review.

**Review by WaffleLapkin — APPROVED** [review]
Thanks!

**Comment by WaffleLapkin** [comment]
@jfgoog you need to find a way to bless the test in such a way, it passes both in CI and locally (this might be non-trivial, given that the test includes paths...)

**Comment by jfgoog** [comment]
Done. I'm not able to test locally, since I'm building on Linux and using llvm-dlltool, which produces different output, which is the entire reason for this change.

Is there a way to trigger the build that failed without sending to submit queue?

**Comment by Nilstrieb** [comment]
>:(
@bors retry

**Comment by Nilstrieb** [comment]
@bors retry
bors please

**Comment by Nilstrieb** [comment]
this is a known spurious issue that happens sometimes, but somehow your PR makes it very angry and triggers it all the time. I'm sorry but it looks like we can't merge the PR right now until it's fixed.
See #113065 for a tracking issue of this problem

**Comment by Nilstrieb** [comment]
@bors r=WaffleLapkin
Thank you for your PR, it accidentally contributed to fixing an LLVM MIPS backend issue!

**Comment by Nilstrieb** [comment]
This looks like a legitimate failure, probably needs some blessing.

**Comment by jfgoog** [comment]
OK, I *think* it's fixed, but I unfortunately don't have a good way to test locally, as I mentioned previously.

**Pull request merged**
